### PR TITLE
Test and modify component

### PR DESCRIPTION
**Pull request by blaskodaniel**
Add test files, modify components and modify test environment

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,66 +31,68 @@
...
   "devDependencies": {
     "@types/enzyme": "^3.9.3",
-    "@types/jest": "^24.0.13",
+    "@types/jest": "^24.0.18",
     "@types/lodash.orderby": "^4.6.6",
-    "@types/react": "^16.8.19",
-    "@types/react-dom": "^16.8.4",
+    "@types/react": "^16.9.2",
+    "@types/react-dom": "^16.9.0",
+    "@types/react-redux": "^7.1.2",
```
Comment (zoltanbedi) [inline package.json]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -30,66 +31,68 @@
...
+    "@types/redux": "^3.6.0",
```
Comment (zoltanbedi) [inline package.json]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -30,66 +31,68 @@
...
-    "@material-ui/core": "^4.4.0",
-    "@material-ui/icons": "^4.2.1",
+    "@material-ui/core": "^4.4.2",
+    "@material-ui/icons": "^4.4.1",
     "@sensenet/client-core": "^2.2.0",
-    "@sensenet/controls-react": "^3.1.0",
-    "@sensenet/default-content-types": "^2.0.0",
+    "@sensenet/controls-react": "^3.1.1",
+    "@sensenet/default-content-types": "^2.0.1",
     "@sensenet/redux": "^5.1.12",
```
Comment (zoltanbedi) [inline package.json]:
```suggestion
```

File: package.json [hunk 4]
```diff
@@ -30,66 +31,68 @@
...
     "@sensenet/repository-events": "^1.4.4",
+    "@testing-library/react": "^9.1.4",
```
Comment (zoltanbedi) [inline package.json]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -30,66 +31,68 @@
...
+    "install": "^0.13.0",
```
Comment (zoltanbedi) [inline package.json]:
```suggestion
```

File: src/components/edit-dialog-body.tsx [hunk 6]
```diff
@@ -36,19 +33,16 @@ const EditPropertiesDialogBody: React.FunctionComponent<{
...
         {(content: CalendarEvent) =>
           content.Id !== ConstantContent.PORTAL_ROOT.Id && (
             <>
               <DialogTitle>Edit</DialogTitle>
               <DialogContent>
                 <EditView
-                  content={content as ContentType}
-                  repository={repo}
+                  content={content as any}
+                  repository={repo as any}
```
Comment (zoltanbedi) [inline src/components/edit-dialog-body.tsx]: Please update `@sensenet/client-core` and remove the any from these places.

File: src/hooks/use-selection-service.ts [hunk 7]
```diff
@@ -1,8 +0,0 @@
-import { useContext, useState } from 'react'
-import { InjectorContext } from '../context/injector-context'
-import { SelectionService } from '../services/selection-service'
```
Comment (zoltanbedi) [inline src/hooks/use-selection-service.ts:3]: Please also remove the services folder.

File: src/components/new-dialog-body.tsx [hunk 8]
```diff
@@ -20,7 +20,7 @@ const NewDialogBody: React.FunctionComponent<{
       <DialogContent>
         <NewView
           handleCancel={handleClose}
-          repository={repo}
+          repository={repo as any}
```
Comment (zoltanbedi) [inline src/components/new-dialog-body.tsx]: Please update @sensenet/client-core and remove the any from these places.

File: test/add-new-event.test.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import { mount, shallow } from 'enzyme'
+import React from 'react'
+import Fab from '@material-ui/core/Fab'
+import AddNewEvent from '../src/components/add-new-event'
+import { SharedContext } from '../src/context/shared-context'
+
+jest.mock('../src/context/shared-context')
```
Comment (zoltanbedi) [inline test/add-new-event.test.tsx]:
```suggestion
```

You don't need to mock this.

File: test/login-form.test.tsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+  })
+
+  it('Password change test', async () => {
+    const wrapper = mount(<LoginForm {...testprop} />)
+    act(() => {
+      ;(wrapper
+        .find(TextField)
+        .at(1)
+        .prop('onChange') as any)({ target: { value: 'testpassword' } } as any)
+    })
```
Comment (zoltanbedi) [inline test/login-form.test.tsx]: What is the expectation here?

File: test/mainpanel.test.tsx [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    await act(async () => {
+      wrapper = mount(
+        <RepositoryContext.Provider value={repo as any}>
+          <SharedContext.Provider value={shareobject as any}>
+            <MainPanel />
+          </SharedContext.Provider>
+        </RepositoryContext.Provider>,
+      )
+    })
+    expect(wrapper).toMatchSnapshot()
```
Comment (zoltanbedi) [inline test/mainpanel.test.tsx:46]: Please rewrite or add test that checks for listItems.

File: test/selection-service.test.tsx [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { Injector } from '@furystack/inject'
+import { ConsoleLogger } from '@furystack/logging'
+import { SelectionService } from '../src/services/selection-service'
```
Comment (zoltanbedi) [inline test/selection-service.test.tsx]: Please remove this test.

File: test/new-dialog-body.test.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+        <RepositoryContext.Provider value={repo as any}>
+          <SharedContext.Provider value={shareobject as any}>
+            <NewDialogBody {...(testprops as any)} />
+          </SharedContext.Provider>
+        </RepositoryContext.Provider>,
+      )
+    })
+
+    const form = wrapper.update().find('button[type="submit"]')
+    console.log(form.html())
```
Comment (zoltanbedi) [inline test/new-dialog-body.test.tsx]:
```suggestion
```

File: test/repository-provider.test.tsx [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    let wrapper: any
+    await act(async () => {
+      wrapper = mount(<RepositoryProvider />)
+    })
+    await act(async () => {
+      ;(wrapper
+        .find(LoginForm)
+        .first()
+        .prop('onLogin') as any)({ username: 'testuser', password: 'testpass', repoUrl: 'http://test.com' } as any)
+    })
```
Comment (zoltanbedi) [inline test/repository-provider.test.tsx]: Need to expect something here.

**Review by zoltanbedi — CHANGES_REQUESTED** [review]

**Comment by zoltanbedi** [comment]
All in all, good job 🎉. I've suggested some changes and asked some questions.

File: test/login-form.test.tsx (hunk 10 above)
Comment (blaskodaniel) [reply test/login-form.test.tsx]: line 20-25 is a change event on the Textfield. I pass the { target: { value: 'testusername' }} object. Then I expect the Textfield value. Did you think of this?

Comment (blaskodaniel) [reply test/login-form.test.tsx]: Oh... I left it there. I will remove it

**Review by zoltanbedi — APPROVED** [review]

**Review by zoltanbedi — APPROVED** [review]

**Pull request merged**